src/ucs/stats/libstats.h

```diff
@@ -56,14 +59,27 @@ struct ucs_stats_class {
 
 /* In-memory statistics node */
 struct ucs_stats_node {
-    ucs_stats_class_t    *cls;
-    ucs_stats_node_t     *parent;
-    char                 name[UCS_STAT_NAME_MAX + 1];
-    ucs_list_link_t      list;
-    ucs_list_link_t      children[UCS_STATS_CHILDREN_LAST];
-    ucs_stats_counter_t  counters[];
+    ucs_stats_class_t        *cls;
+    ucs_stats_node_t         *parent;
+    char                     name[UCS_STAT_NAME_MAX + 1];
+    ucs_list_link_t          list;
+    ucs_list_link_t          children[UCS_STATS_CHILDREN_LAST];
+    ucs_list_link_t          type_list;
+    ucs_stats_filter_node_t  *filter_node;
+    ucs_stats_counter_t      counters[];
 };
 
+struct ucs_stats_filter_node {
```

Comment:
what is the purpose of this node? can you add some doc please?
maybe is should be renamed to ucs_stats_agg_node_t?
also, seems like many fields are common with ucs_stats_node , can we create a common struct to avoid code duplication?